In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly_express as px
import plotly.graph_objects as go

- Jag kommer jämföra andelen infekterade av covid-19 i varje kontinent.
- Vad jag misstänker är följande order (lägst till störst andel): Oceania -> Europa -> Asia -> America -> Africa
- Jag tänker mig att Oceania kommer först eftersom dem är väldigt utspridda, många öar, samt har många i-länder i sig.
- Jag tror att Europa kommer 2:a eftersom vi har många i-länder och generellt sett en vetenskaplig kultur. 
- Asien tänker jag mig kan komma trea, alltså före america eftersom länder som USA har en kultur av icke-vaccinerande och konspirationstänk i vissa grupper. Jag har svårt att gissa på hur resten av America kommer påverka den slutliga andelen. Asien har även länder som Kina, vilket gör det svårt att lita exakt på att deras siffror av infekterade stämmer helt.
- Jag gissar på att Afrika kommer sist endast för att dem har såpass många u-länder vilket gör det svårt att sätta stopp för mycket infektioner. Dock kan jag ha helt fel här eftersom Afrika är hem till många stammar som tex lever isolerat från resten av populationen.

In [2]:
worldcoviddata = pd.read_csv("../Data/worldwidedata_covid.csv")

def continent_total(variable_name, continent_name):
    variable_name = worldcoviddata[(worldcoviddata["country"] == continent_name)].sort_values(by='cumulative_count', ascending=False).head()
    print(variable_name['cumulative_count'].head(1))

In [3]:
continent_total(worldcoviddata, "Europe (total)")

16595    216693251.0
Name: cumulative_count, dtype: float64


- Europa totalt: 216,693,251 antal fall
- Europa total befolkning: 747,636,000 (2020)
- Europa andel infekterade: 28.98%

In [4]:
continent_total(worldcoviddata, "Africa (total)")

380    11937718.0
Name: cumulative_count, dtype: float64


- Afrika totalt: 11,937,718 antal fall
- Afrika total befolkning: 1,340,000,000 (2020)
- Afrika andel infekterade: 0.89% infekterade

In [5]:
continent_total(worldcoviddata, "Oceania (total)")

36504    9196406.0
Name: cumulative_count, dtype: float64


- Oceania totalt: 9,196,406 antal fall
- Oceania total befolkning: 42,677,000 (2020)
- Oceania andel infekterade: 21.54% infekterade

In [6]:
continent_total(worldcoviddata, "America (total)")

1124    160090765.0
Name: cumulative_count, dtype: float64


- America totalt: 160,090,765 antal fall
- America total befolkning: 1,002,000,000 (2020)
- America andel infekterade: 15.97% infekterade

In [7]:
continent_total(worldcoviddata, "Asia (total)")

3102    135192948.0
Name: cumulative_count, dtype: float64


- Asien totalt: 135,192,948 antal fall
- Asien total befolkning: 4,641,000,000 (2020)
- Asien andel infekterade: 2.91% infekterade
----------------------------------
- Slutgiltiga ordningen (Lägst till högst)
- Africa --> Asien --> America --> Oceania --> Europa
- Det var allså i stort sett tvärt emot vad jag tänkte mig att det skulle vara, dock verkar det otroligt suspekt att inte ens 1% av Afrika har infekterats medans nästan 30% infekterats i Europa. Det hade inte förvånat mig om det rapporterades sämre i dem kontinenterna som hade såpass låg andel.
- Jag tänkte mig först när jag såg det här resultatet att det kanske har något med befolkningstätheten i varje kontinent, alltså att Europa har hög befolkningstäthet(69.7 pers per km^2). Det stämmer också, dock så har Asien ännu högre (86,7 pers per km^2). Det kan alltså inte vara den enda viktiga variabeln i spel till varför Europa har högst andel.
- Jag kan dock säga att efter att ha kollat upp andelen i varje kontinent så är det absolut inte tillräckligt för att dra några slutsatser kring varför andelen ser ut som dem gör.

In [8]:
worldcoviddata

country country_code continent  population indicator  weekly_count  \
0      Afghanistan          AFG      Asia    38928341     cases           0.0   
1      Afghanistan          AFG      Asia    38928341     cases           0.0   
2      Afghanistan          AFG      Asia    38928341     cases           0.0   
3      Afghanistan          AFG      Asia    38928341     cases           0.0   
4      Afghanistan          AFG      Asia    38928341     cases           0.0   
...            ...          ...       ...         ...       ...           ...   
53069       Cyprus          CYP    Europe      896007    deaths          20.0   
53070       Cyprus          CYP    Europe      896007    deaths           9.0   
53071       Cyprus          CYP    Europe      896007    deaths           NaN   
53072       Cyprus          CYP    Europe      896007    deaths           0.0   
53073       Cyprus          CYP    Europe      896007    deaths           5.0   

      year_week  rate_14_day  cumulative_count  \
0       2020-01          NaN               0.0   
1       2020-02     0.000000               0.0   
2       2020-03     0.000000               0.0   
3       2020-04     0.000000               0.0   
4       2020-05     0.000000               0.0   
...         ...          ...               ...   
53069   2022-20    44.642508            1209.0   
53070   2022-21    32.365819            1218.0   
53071   2022-22          NaN               NaN   
53072   2022-23          NaN               NaN   
53073   2022-24     5.580314               NaN   

                                    source  note  
0      Epidemic intelligence national data   NaN  
1      Epidemic intelligence national data   NaN  
2      Epidemic intelligence national data   NaN  
3      Epidemic intelligence national data   NaN  
4      Epidemic intelligence national data   NaN  
...                                    ...   ...  
53069                       TESSy COVID-19   NaN  
53070                       TESSy COVID-19   NaN  
53071                       TESSy COVID-19   NaN  
53072                       TESSy COVID-19   NaN  
53073                       TESSy COVID-19   NaN  

[53074 rows x 11 columns]